balancer/weightedroundrobin/balancer.go

```diff
@@ -527,21 +592,35 @@ func (w *weightedSubConn) updateConnectivityState(cs connectivity.State) connect
 
 // weight returns the current effective weight of the subconn, taking into
 // account the parameters.  Returns 0 for blacked out or expired data, which
-// will cause the backend weight to be treated as the mean of the weights of
-// the other backends.
-func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration) float64 {
+// will cause the backend weight to be treated as the mean of the weights of the
+// other backends. If forScheduler is set to true, this function will emit
+// metrics through the mtrics registry.
+func (w *weightedSubConn) weight(now time.Time, weightExpirationPeriod, blackoutPeriod time.Duration, forScheduler bool) (weight float64) {
 	w.mu.Lock()
 	defer w.mu.Unlock()
+
+	weight = 0
+	if forScheduler {
+		defer endpointWeightsHandle.Record(w.metricsRecorder, weight, []string{w.target, w.locality}...)
```

Comment:
You shouldn't ever do
```go
f(x, y, []string{a, b}...)
```

It should be:

```go
f(x, y, a, b)
```

Reply:
Ah sounds good noted. Switched for all callsites.